docassemble/assemblylinewizard/data/questions/assembly_line.yml

```diff
@@ -909,55 +957,61 @@ need:
   - interview_label
   - attachment_variable_name
 code: |
-  preview_question = interview.questions.appendObject()
-  preview_question.type = 'interstitial'
-  preview_question.continue_button_field = interview_label + '_preview_question' 
-  preview_question.question_text = 'Review your form before you sign it'  
-  preview_question.subquestion_text = "Here is a preview of the form you will sign on the next page. \n\n ${ al_user_bundle.as_pdf(key='preview') }\n\n"
-  preview_question.subquestion_text += "Click the image to open it in a new tab. Click the 'Make changes' button to fix any mistakes.\n\n"
-  preview_question.subquestion_text += "${ action_button_html(url_action('review_" + interview_label + "'), label='Make changes', color='info') } \n\n"
-  preview_question.subquestion_text += "Remember to come back to this window to continue and sign your form."
-  preview_question.is_mandatory = False  
-
-  created_preview_question = True
+  preview_question = interview.blocks.appendObject()
+  preview_question.template_key = 'preview screen'
+  preview_question.data = {
+    "interview_label": interview_label
+  }
+  add_preview_question = True
 ---
 need:
   - attachments
   - attachment_variable_name
-  - interview
   - interview_label
+  - attachment.name
+  - attachment.type
 code: |
   # This is the question that goes with the regular attachment
-  end_question = interview.questions.appendObject() # 'all_done')
-  end_question.id = "download" # Use a consistent ID for Google Analytics
-  end_question.event = interview_label + "_download"
-  end_question.type = 'question'
-  end_question.progress = "100"
-  end_question.question_text = 'Your form is ready to download.'
-
-  # Add a preview of the form to the draft ending screen
-  end_question.subquestion_text = 'Thank you ${users}. Your form is ready to download and deliver.\n\n'
-  end_question.subquestion_text += 'View, download and send your form below. Click the "Make changes" button to fix any mistakes. \n\n'
-  end_question.subquestion_text += "${ action_button_html(url_action('review_" + interview_label + "'), label='Make changes', color='info') } \n\n"
-  end_question.subquestion_text += "${ al_user_bundle.download_list_html() }\n\n"
-  end_question.subquestion_text += "${ al_user_bundle.send_button_html() }\n"
-  end_question.is_mandatory = False
-  end_question.attachment_variable_name = attachment_variable_name
-  end_question.interview = interview
-  end_question.interview_label = interview_label
+  end_question = interview.blocks.appendObject()
+  end_question.template_key = 'download screen'
+  end_question.data = {"interview_label": interview_label}
+  add_end_question = True
+---
+need:
+  - attachments
+  - attachment_variable_name
+  - interview_label
+  - attachment.name
+  - attachment.type
+code: |
+  # TODO: allow multiple attachments here eventually
```

Comment:
A good practice I've picked up is to use `TODO(issue number)` (so TODO(#113) here). I don't do it much, but feel like it makes sense in this big refactor.